lib/src/internal/shard/shard_manager.dart

```diff
@@ -137,16 +162,18 @@ class ShardManager implements IShardManager {
     }
   }
 
-  void _connect(int shardId) {
-    logger.fine("Setting up shard with id: $shardId");
-
-    if (shardId < 0) {
+  void _connect(List<int> toSpawn) {
+    if (toSpawn.isEmpty) {
       return;
     }
 
+    int shardId = toSpawn.removeLast();
+
+    logger.fine("Setting up shard with id: $shardId");
+
     _shards[shardId] = Shard(shardId, this, connectionManager.gateway);
 
-    Future.delayed(_identifyDelay, () => _connect(shardId - 1));
+    Future.delayed(_identifyDelay, () => _connect(toSpawn));
   }
```

Comment:
was it tested and it works correctly?

Reply:
This has been tested and works correctly